packages/a11y-base/src/aria-id-reference.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright (c) 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { addValueToAttribute, removeValueFromAttribute } from '@vaadin/component-base/src/dom-utils.js';
+
+const setAriaIDReference = (target, attr, newId, oldId) => {
+  if (!target) {
+    return;
+  }
+
+  if (oldId) {
+    removeValueFromAttribute(target, attr, oldId);
+  }
+
+  if (newId) {
+    // TODO: indicate that provided attribute value is managed by this helper.
```

Comment:
Could you explain it more? Why do we need to indicate that?

Reply:
The idea is to distinguish user-originated attribute value from the generated one that is managed by the component itself.

Reply:
Improved the TODO comment accordingly. I'm going to take a stab at the actual solution, too.